subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileToolchainIntegrationTest.groovy

```diff
@@ -391,28 +601,17 @@ public class Foo {
             }
         """
 
-        file("src/main/java/Foo.java") << "public class Foo {}"
-
         when:
-        runWithToolchainConfigured(jdk)
+        withInstallations(jdk).run(":compileJava", "--info")
 
         then:
         outputDoesNotContain("Compiling with Java command line compiler")
         outputContains("Compiling with toolchain '${jdk.javaHome.absolutePath}'.")
         javaClassFile("Foo.class").exists()
+        jdk.javaVersion == JavaVersion.forClass(javaClassFile("Foo.class").bytes)
 
         where:
-        javaVersion << [
-            JavaVersion.VERSION_1_8,
-            JavaVersion.VERSION_1_9,
-            JavaVersion.VERSION_11,
-            JavaVersion.VERSION_12,
-            JavaVersion.VERSION_13,
-            JavaVersion.VERSION_14,
-            JavaVersion.VERSION_15,
-            JavaVersion.VERSION_16,
-            JavaVersion.VERSION_17
-        ]
+        javaVersion << (8..19).collect { JavaVersion.toVersion(it) }
```

Comment:
So we are testing all supported Java versions >= 8? How about filtering `JavaVersion.values()` instead?

Reply:
I assume that was an original intention, just made it a bit more concise.

I wouldn't try to fix this with filtering `JavaVersion` values, given that's implementation is going to change in 8.0 anyways.
- https://github.com/gradle/gradle/issues/22396

Reply:
Never mind, I fixed it and added a comment about the intention to test the range from 8 up to the current version